sdks/python/apache_beam/metrics/metric_test.py

```diff
@@ -162,6 +169,17 @@ def process(self, element):
         outputs_counter.key.metric.name, 'metrics_user_counter_element')
     self.assertEqual(outputs_counter.committed, 4)
 
+    # Verify user distribution counter.
+    metric_results = res.metrics().query()
+    namespace = 'apache_beam.metrics.metric_test.SomeDoFn'
+    hc.assert_that(
+        metric_results['distributions'],
+        hc.contains_inanyorder(
+            MetricResult(
+                MetricKey('ApplyPardo', MetricName(namespace, 'element_dist')),
+                DistributionResult(DistributionData(10, 4, 1, 4)),
+                DistributionResult(DistributionData(10, 4, 1, 4)))))
```

Comment:
not all runners support attempted metrics, please only compare against committed metrics for now

Reply:
I see. Thanks for the information! 

I checked MetricResult interface, it only works when passed in both 'commited' and 'attempted'. 

I will take a look what alternative we need to do.  If you by any chance are aware of a particular API I should use, please let me know.

Reply:
Write your own matcher.

Reply:
Sure. 

Done update. 

Please take a look if it makes sense.